content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+
+In the last summers we did face the challenge, that the initially planned cooling capacity was not sufficient anymore for the new generations of server and storage that we need to deploy in the CTC. As a consequence the lab temperature was rising to a level, that caused several incidents, where HPE ProLiant servers were automatically shutting down to avoid damages. In order to avoid this situation we were looking for a CTC lab temperature warning system that is easy to implement. An heads up of the rising temperature in the CTC would allow us to shutdown equipment that is currently not urgently needed for customer demos - helping us to ensure that the booked customer demos could still run.
+
+# Redfish® iLO RESTful API as a solution?
+
+Each HPE ProLiant server has a lot of temperature sensors spread across the system board, CPU, memory and power supplies. This detailed temperature data is available and visualized on each ILO interface in the Power&Thermal section:
+
+![](/img/ilotemperature.png "ILO Temperature Information")
+
+The first sensor, 01-Inlet Ambient temperature, is measuring the inlet temperature at the front of the server and can be used as an indicator of the CTC temperature.  Now that I have identified a usable data point an automated reading of the temperature values is needed. Having used the SimpliVity REST API to build the SimpliVity-Prometheus connector, that is used to monitor the core infrastructure, it was a natural step for me to look into the [Redfish® iLO RESTful API](https://developer.hpe.com/platform/ilo-restful-api/home/). The Redfish® API ecosystem is an open industry-standard specification (published by the Distributed Management Task Force [DTMF](http://www.dmtf.org/standards/redfish)) and provides remote server provisioning, inventory and monitoring.
+
+A directory of the available Redfish® resources at an interface can be retrieved with the following RESTful API command: 
+
+\`GET {{baseUrl}}/redfish/v1/resourcedirectory\`
+
+I used [Postman](https://www.postman.com/) as an developer tool to retrieve and search the resource directory. The above RESTful API comand was used in Postman with a basic username/password authorization at one of the ILOs in the CTC, where the baseUrl is defined as <https://iloIPaddress>. I searched in the obtained resource directory for "thermal" in order to find the correct command to retrieve thermal data:
+
+![](/img/redfishresourcedirectory.png "Redfish resource directory (Postman output)")
+
+Hence, I can retrieve thermal server data with the following RESTful command:
```

Comment:
Remember that most of your blog post is written in the past tense. Therefore, this should read something more like:

"I found that I could retrieve thermal server data using the following RESTful command:"